### PR TITLE
Update to repo-config v0.7.1

### PR DESCRIPTION
**Pull request by llucax**
- ci: Add extra debug information.
- ci: Add meta-jobs (`nox-all` and `nox-cross-arch-all`) to be able to require them in branch protection rules so we don't have to update the protection rules each time we add or remove a job from the matrix.
- ci: Add support for cross-arch testing, including testing in arm64.
- ci: Ignore dependabot pushes (they are tested via the PR).
- ci: Run `nox` sessions concurrently to speed up tests.
- ci: Test installation of the package in different platforms.
- dependabot: Update dependencies in groups, so we get only one PR to update multiple dependencies.
- dependabot: Update dependencies monthly instead of daily.
- docs: Add a few handy `mkdocs`/`markdown` extensions.
- docs: Add `mkdocs-macros-plugin` to be able to use macros in the documentation.
- docs: Fix formatting of mermaid diagrams.
- docs: Move some support files for the documentation website to names starting with `_` to make it more clear they are only support files.
- docs: Replace the `mkdocs-section-index` plugin (which has caused problems in the past) with the `mkdocs-material` built-in `navigation.indexes` feature.
- docs: Show inherited class members in the documentation.
- docs: Use a custom style to show code annotation numbers for better ordering.
- docs: Use the new documentation website versioning scheme. This means now multiple development branches are exposed, as well as pre-releases. Also the order of the versions is improved. The `next` version is replaced by the multiple `vX.Y-dev` versions.
- nox: Add a `flake8` session, mainly needed to run `pydocstyle` but also use it to run `pycodestyle` and `flake8` regular checks. `flake8` is much faster than `pylint`, so when there are conflicting checks we prefer `flake8`.
- nox: Replace `darglint` with `pydoclint`.
- Remove unnecessary fields from the cookiecutter replay file.
- Unify some `pyproject.toml` options and move other tool options (like `mypy`) to this file.
- Remove types from the docs
- Remove empty lines between docstrings and code
- Remove double comment char (`#`)
- Fix arguments documentation
- Add missing `Yields` section to the documentation
- Add missing `Returns` section to `consume()`
- Ignore `Raises` sections without explicit `raise` statements
- Remove obsolete `darglint` ignore directives

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -154,3 +154,30 @@ These are the steps to create a new release:
...
 7. Celebrate!
+
+##  Cross-Arch Testing
+
+This project has built-in support for testing across multiple architectures.
+Currently, our CI conducts tests on `arm64` machines using QEMU emulation. We
+also have the flexibility to expand this support to include additional
+architectures in the future.
+
+This project containers Dockerfiles that can be used in the CI to test the
```
Comment (daniel-zullo-frequenz) [inline CONTRIBUTING.md]: hmm I thought we fixed it in repo-config, it should be `contains` instead of `containers`

File: docs/_css/style.css [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+/* More space at the bottom of the page */
+.md-main__inner {
+  margin-bottom: 1.5rem;
+}
+
+/* Code annotations with numbers.
+ *
+ * Normally annothations are shown with a (+) button that expands the
```
Comment (daniel-zullo-frequenz) [inline docs/_css/style.css]: typo that might need to be fixed in repo-config
```suggestion
 * Normally annotations are shown with a (+) button that expands the
```

**Review by daniel-zullo-frequenz — DISMISSED** [review]
Just a couple of typos, LGTM

**Comment by daniel-zullo-frequenz** [comment]
I removed the auto-merge as I wasn't sure if you might want to fix the typos or not. Anyway feel free to merge as it is

File: CONTRIBUTING.md (hunk 1 above)
Comment (llucax) [reply CONTRIBUTING.md]: Yeah, I just didn't regenerate the templates yet :(

I'll do it later today.

**Review by shsms — APPROVED** [review]

**Pull request merged**